src/main/java/seedu/address/model/team/Team.java

```diff
@@ -0,0 +1,67 @@
+package seedu.address.model.team;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.List;
+
+import seedu.address.model.contact.Contact;
+import seedu.address.model.contact.Name;
+import seedu.address.model.contact.UniqueContactList;
+
+public class Team {
+    private final Name name;
+    private final UniqueContactList members = new UniqueContactList();
+
+    public Team(Name name) {
+        requireNonNull(name);
+        this.name = name;
+    }
+
+    public Team(Name name, List<Contact> members) {
+        requireAllNonNull(name, members);
+        this.name = name;
+        this.members.setContacts(members);
+    }
+
+    public Name getName() {
+        return name;
+    }
+
+    public List<Contact> getMembers() {
+        return members.asUnmodifiableObservableList();
+    }
+
+    /**
+     * Returns true if both teams have the same name.
+     * This defines a weaker notion of equality between two teams.
+     */
+    public boolean isSameTeam(Team otherTeam) {
+        if (otherTeam == this) {
+            return true;
+        }
+
+        return otherTeam != null
+                && otherTeam.getName().equals(getName());
+    }
+
+    /**
+     * Returns true if both teams have the same identity and data fields.
+     * This defines a stronger notion of equality between two teams.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof Team)) {
+            return false;
+        }
+
+        Team otherTeam = (Team) other;
+        return name.equals(otherTeam.name)
```

Comment:
Let name be the sole unique identifier for teams. Using name AND team members as unique identifier of teams can cause issues down the line, when suddenly adding a member to a team causes it to be non unique with another team in the teamlist

Reply:
`Team::equals` and `Team::isSameTeam` mostly were copied over from Contact, so the same issue should apply to there too. However, I feel that you have the issue the wrong way around. `Team::isSameTeam` is currently used by `UniqueTeamList` to ensure that teams have unique names. If we instead changed `Team::equals` to follow the same logic, we might run into problems in the future. For example, if we wanted to relax the restriction on teams not having the same name, then `Team::equals` would not work to identify unique teams. I also think that having `Team::equals` use a stricter equality is necessary in the case of tests, where we might want to verify if a `Team` object contains the correct members.